FLOE-506: update to new Hewlett logo in a responsive layout

waharnum opened the pull request:
@jobara, if you have time to look at this one today as discussed in planning meeting. Thanks!

waharnum commented:
I also added a fix for the image width of the front page on small screens like the iPhone 5.

jobara commented on news/2014-04-21-metadata-search.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: Metadata Search | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
Should probably put the `<meta>` tag on a separate line.

jobara commented on news/2014-09-30-lumen-learning.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: Lumen Learning | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move `<meta>` to a separate line.

jobara commented on news/2015-01-08-oer-a11y-sprint.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: OER Accessibility Sprint | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on news/2015-03-26-uio-wp-plugin.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: Learner Options WordPress Plugin | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on news/2015-05-08-workshop-business-advantages.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: The Business Advantages of Making Special the Norm | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on news/2015-11-16-phet.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: PhET Accessible Physics Simulations | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on news/2016-03-28-upcoming-talks-on-inclusive-digital-education.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: Upcoming Talks on Inclusive Digital Education | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on news/2016-11-30-DEEP-conference-2016.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: DEEP Conference 2016 | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on news/2016-11-30-IIDEX-2016.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: DEEP Conference 2016 | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on news/2016-11-30-UN-international-day-of-persons-with-disabilities.html:
```diff
@@ -4,7 +4,7 @@
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <title>News: DEEP Conference 2016 | floe</title>
-        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/>
+        <link type="image/x-icon" href="../images/favicon.ico" rel="shortcut icon"/><meta name="viewport" content="width=device-width, initial-scale=1">
```
similar to above, please move <meta> to a separate line.

jobara commented on index.html:
```diff
@@ -280,15 +282,17 @@
-                        </div>
+                                <div class="row">
+                                    <div class="hewlett-logo-container columns large-5 medium-7 small-9">
+                                        <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" x="0" y="0" viewBox="0 0 1057.5 225.3" xml:space="preserve"><path fill="#243C4C" d="M94 225.3h6.8v-44h59.9v-6.8H94v50.8zm-17 0h6.8v-93H94v32h66.7v-6.8h-60v-25.2h59.9v-6.8h-59.9v-10.3h59.9v-6.8H77v116.9zm6.7-110H94v10.3H83.7v-10.3zm114.5-26l4.5-17.5 4.5 17.5h4.7l8.2-29.3h-5.6l-4.9 18.8-4.8-18 ...
```
If we aren't going to style the SVG with the contrast themes, it might make sense to not embed the SVG directly. The reason is that it's error prone to potential accidental modifications or copy and paste errors. Also, I don't believe it will be cached by the browser for use across pages.

There might be some options from the research or linked articles for svgs icons https://wiki.fluidproject.org/display/fluid/Research+SVG+Icons

waharnum replied:
I think we could potentially make the SVG adaptable to the context themes, but not without some work (and checking with Hewlett before modifying their logo in various ways).

Based on your research the best option is probably to use it as the source of an image tag - I believe we are still officially supporting IE11. I'll look at this option.

waharnum replied:
Actually, I see we have various contrast theme versions of the old Hewlett logo in the `images/hewlett/` directory, but the current floeproject.org doesn't change the logo to the various contrast themes. I wonder if this is functionality that was lost somewhere? It doesn't appear from the current style sheet that it has existed in some time.

waharnum replied:
I ended up implementing contrast styles for the new Hewlett logo, since this is easy to do with SVG.

Is there any alternative to `<use>` if we want to style with CSS?

jobara replied:
From what I recall, the alternative would be to embed the svg.

waharnum replied:
OK. I believe we still try to support IE11, so if `<use>` doesn't work for that browser, I think we do have to embed the SVG directly on the page. 

Upside: the contrast implementations are really easy with an SVG image.

jobara commented on index.html:
```diff
@@ -280,15 +282,17 @@
+                                    <div class="hewlett-logo-container columns large-5 medium-7 small-9">
+                                        <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" x="0" y="0" viewBox="0 0 1057.5 225.3" xml:space="preserve"><path fill="#243C4C" d="M94 225.3h6.8v-44h59.9v-6.8H94v50.8zm-17 0h6.8v-93H94v32h66.7v-6.8h-60v-25.2h59.9v-6.8h-59.9v-10.3h59.9v-6.8H77v116.9zm6.7-110H94v10.3H83.7v-10.3zm114.5-26l4.5-17.5 4.5 17.5h4.7l8.2-29.3h-5.6l-4.9 18.8-4.8-18 ...
+                                    </div>
+                                    <div class="floe-hewlett-text columns large-7 medium-9 small-9"><p>
```
Nesting the tag names would probably help with readability.

jobara replied:
That is, putting the `<p>` tag on a new line.

jobara commented on news/2015-05-08-workshop-business-advantages.html:
```diff
@@ -116,15 +116,17 @@
+                                    </div>
+                                    <div class="floe-hewlett-text columns large-7 medium-9 small-9"><p>
+                                    Funded by a grant from <a href="http://www.hewlett.org">The William and Flora Hewlett Foundation</a>.</p></div>
+                                </div>
```
The indentation seems to be off.

jobara replied:
Similar in other HTML files.

jobara commented on css/style.css:
```diff
@@ -507,33 +500,86 @@ address {
+.fl-theme-lgdg .floe .floe-image-videoPlayer     { background-image: url("../images/lgdg-floe-03.png"); }
+
+.fl-theme-bw .floe .hewlett-logo-container svg {
+    border: 1px solid #000;
```
You might be able to simplify all of the contrast themes by using [currentColor](https://developer.mozilla.org/en/docs/Web/CSS/color_value#currentcolor_keyword). You'd then be able to have the same declaration for all of the themes.

waharnum replied:
Good tip, works great - didn't know `currentColor` was a thing. Simplifies them considerably.

jobara commented on css/style.css:
```diff
@@ -507,33 +500,54 @@ address {
+    border: 1px solid currentColor;
+}
+
+.fl-theme-bw .floe .hewlett-logo-container svg path, .fl-theme-wb .floe .hewlett-logo-container svg path, .fl-theme-by .floe .hewlett-logo-container svg path, .fl-theme-yb .floe .hewlett-logo-container svg path, .fl-theme-lgdg .floe .hewlett-logo-container svg path, .fl-theme-bw .floe .idrc-logo-container svg path, .fl-theme-wb .floe .idrc-logo-container svg path, .fl-theme-by .floe .idrc-logo-co ...
```
instead of having all the selectors on a single line, could you add line breaks after each comma. At this length I'm finding it hard to read and see what all the selectors are. Same for the other contrast theme related declaration above.

jobara commented on index.html:
```diff
@@ -280,15 +282,32 @@
+                                <div class="idrc-logo-container columns large-5 medium-7 small-9">
+                                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 375.87 77.161"><title>IDRC Logo</title><g fill="#00A0CC"><path d="M21.701 12.809c6.365 10.428 14.345 23.502 19.237 34.619 2.625-3.069 6.783-7.631 10.449-11.338V10.161H20.086l1.615 2.648z"></path><path d="M42.243 50.534c1.541 3.866 2.62 7.419 3.018 10.443.454 3.457-.195 6.477-1.783 8.285-.965 1.102-2.21 1.684- ...
+                                    <text x="70" y="35" font-family="Helvetica" font-size="37" font-weight="700">inclusive design</text>
+                                    <text x="80" y="72" font-family="Helvetica" font-size="37" font-weight="700">research centre</text>
```
We should move these inline styles to css.

waharnum replied:
There is a case to be made for an SVG logo for keeping styles like `font-family` inline - it means the logo markup is a self-contained "instance" of the logo and can be used elsewhere without change in appearance. Moving some of the styles to the CSS will mean this self-containment goes away.

I don't have a strong feeling about this, but want to make the opposing case.

jobara replied:
@waharnum that's fine, I was thinking that was outside of the svg.

jobara commented:
@waharnum noticed that the border separating the footer from the content is missing for the light grey on dark grey contrast theme.

![screen shot 2017-04-27 at 12 36 12 pm](https://cloud.githubusercontent.com/assets/556024/25493983/2e1522e0-2b46-11e7-9fc4-4811f3c77b63.png)

waharnum commented:
I think this is ready for more review, @jobara

jobara commented:
@waharnum there is a small bug on the Hewlett logo in the contrast mode, where there is a filled in square on the H.

Default Style:
<img width="583" alt="screen shot 2017-04-30 at 9 36 20 pm" src="https://cloud.githubusercontent.com/assets/556024/25569924/3703790c-2ded-11e7-8e8a-f74c28bf91cb.png">

Contrast Applied:
<img width="575" alt="screen shot 2017-04-30 at 9 36 31 pm" src="https://cloud.githubusercontent.com/assets/556024/25569927/3f6ce998-2ded-11e7-9421-6c46a80e6576.png">

jobara commented:
In firefox the top dot of the IDRC logo is cropped.

<img width="461" alt="screen shot 2017-04-30 at 9 38 26 pm" src="https://cloud.githubusercontent.com/assets/556024/25569932/6b70350e-2ded-11e7-8e32-bdf80031fbe9.png">

jobara commented:
The logos are misaligned in IE 11 on Win 7, in the "desktop" view.

<img width="989" alt="screen shot 2017-04-30 at 9 42 09 pm" src="https://cloud.githubusercontent.com/assets/556024/25569985/ff5d9c66-2ded-11e7-9c7c-a480e0dcbb76.png">

waharnum commented:
@jobara I think I've addressed all the above issues (there unfortunately doesn't seem to be a non-hacky way of making IE11 size SVGs properly, so I went with IE-specific styles) - there's some discussion at https://css-tricks.com/scale-svg/ about various tricks, but since IE11 is the outlier, I'd prefer to treat it as such.

jobara commented on index.html:
```diff
@@ -278,22 +280,269 @@
+                                    	c-1.113,0-1.995,0.381-2.644,1.143c-0.65,0.762-1.028,1.87-1.136,3.325h7.207c0-0.664-0.065-1.272-0.198-1.824
+                                    	c-0.131-0.552-0.336-1.023-0.615-1.414c-0.277-0.391-0.632-0.693-1.062-0.908S289.307,77.61,288.701,77.61z"/>
+                                    <use xlink:href="#idrc_symbol"  width="116" height="148" x="-58" y="-74" transform="matrix(0.5 0 0 -0.5 54 60)" overflow="visible"/>
+                                    </svg>
```
Could you adjust the indentation so that the contents of the svgs is indented one level.

jobara commented on index.html:
```diff
@@ -278,22 +280,269 @@
-                        </div>
+                            <div class="row align-center idrc-credit-row">
+                                <div class="idrc-logo-container columns large-5 medium-7 small-9">
+                                    <title>IDRC Logo</title>
```
Is it valid to use a `<title>` element outside of the `<head>`? Could you also run all of the files through the HTML validator to make sure there aren't any issues.

waharnum replied:
No, that should be inside the SVG markup. Fixed.

waharnum replied:
Ran through the validator and fixed an ID duplication issue on the SVGs. There were aslo some markup complaints with the UIO markup on the page due to duplicate IDs between the smallScreen and wideScreen markup. I've updated the ids to be distinct without any noticeable impact, but want to note this - I don't think the IDs have any implications because all events and DOM binding should be based on the classes, correct?

jobara replied:
@waharnum, those UIO id's for show-hide and reset shouldn't be necessary. It's also a bug in Infusion, but I've removed them in the branch I'm currently working on.

jobara commented:
@waharnum the light grey on dark grey them doesn't seem to be applying to the hewlett logo.
<img width="539" alt="screen shot 2017-05-01 at 1 56 46 pm" src="https://cloud.githubusercontent.com/assets/556024/25588375/0a74f04e-2e76-11e7-8137-f2cb7b500216.png">

jobara commented:
@waharnum there seems to be an issue with the logos in the black on white theme.

<img width="523" alt="screen shot 2017-05-01 at 1 56 03 pm" src="https://cloud.githubusercontent.com/assets/556024/25588398/2147876e-2e76-11e7-9209-668988aec62d.png">

waharnum commented:
@jobara Messed up some of the theme CSS when I was fixing another issue - sorry for not catching it. Should now be fixed.

waharnum commented:
How does this look now @jobara?

jobara commented on css/style.css:
```diff
@@ -507,33 +505,149 @@ address {
+/* IE-only hacks for better SVG sizing */
+
+/* Large, two column */
+@media all and (min-width: 64em) and (-ms-high-contrast:none)
```
@waharnum a potential issue with this approach is that if the user has made use of contrast themes in Windows; which is a possibility for a user who may be setting contrast preferences in UIO. In that case, the height won't be applied properly.

https://msdn.microsoft.com/en-us/library/hh771830(v=vs.85).aspx

jobara replied:
@waharnum just bumping the comment about using both `active` and `none` in the media query to ensure that the height is applied when a contrast theme is in use on Windows.

jobara commented on css/style.css:
```diff
@@ -507,33 +505,149 @@ address {
+/* IE-only hacks for better SVG sizing */
+
+/* Large, two column */
+@media all and (min-width: 64em) and (-ms-high-contrast:none)
```
@waharnum this solution that uses both `active` and `none` seems to work.

https://paper-leaf.com/blog/2014/09/targeting-ie-10-11-browsers-css/

waharnum commented:
An update on trying to make use of SVG `use` - I've had a real struggle getting this to respond cross-browser (even just on Mac OS) to CSS in a consistent, non-hacky way. 

Some of the issues are described at https://css-tricks.com/svg-use-with-external-reference-take-2/ - I've also seen the issues documented at http://monopine.com/svg-demo/

The Hewlett logo in particular has been an issue because it uses two separate `path` fill colors (and a third transparent path fill to "cut out" part of the logo). If the SVG is inline, I can address the separate `path` classes directly in the contrast styles, but this doesn't seem possible cross-browser with `use`.

At the moment, the consensus on `use` seems to be:
- cross-browser CSS behaviour is not consistent in terms of being able to address individual elements 
- the best solutions involve use of cascading `inherit` or `currentColor`, and seem to require the removal of all attribute styles to work.

Can we live with the SVGs inlined for now? I don't feel like `use` has reached a point of maturity in non-FF browsers to suit the use case here.

jobara commented on css/style.css:
```diff
@@ -507,33 +505,149 @@ address {
+
+/* Small, one column */
+
+  @media all and (max-width: 40em) and (-ms-high-contrast:none)
```
Same comment as above about using `active` and `none`. Also it may be worthwhile to comment if this is specific for IE and/or Edge.

jobara commented on security.html:
```diff
@@ -5,6 +5,7 @@
         <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <link type="image/x-icon" href="images/favicon.ico" rel="shortcut icon"/>
+        <meta name="viewport" content="width=device-width, initial-scale=1">
```
This meta tag duplicates one on line 5.

jobara commented on prefsEditors.html:
```diff
@@ -5,6 +5,7 @@
         <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
         <meta content="width=device-width, initial-scale=1.0" name="viewport">
         <link type="image/x-icon" href="images/favicon.ico" rel="shortcut icon"/>
+        <meta name="viewport" content="width=device-width, initial-scale=1">
```
This meta tag duplicates one on line 5.

jobara commented:
@waharnum was it intentional to remove the CNAME file?

sachin10101998 commented:
Stale and inactive PR. Can be closed.

greatislander commented:
This work will be a helpful basis for new updates so I've labelled it as mothballed for reference.
